File: 2023/aoc-day8-p1.js
// Put directional input here:
let directions = ''.split('');

// Put | separated map input here:
let map = ''.split('|');

let navMap = {};
map.forEach((node) => {
    let navTokens = node.split(' = ');
    let lrTokens = navTokens[1].replace('(', '').replace(')', '').split(', ');
    navMap[navTokens[0]] = {
        L: lrTokens[0],
        R: lrTokens[1]
    };
});

let stepCounter = 0;
let input = 'AAA';
let directionInd = 0;
while (input !== 'ZZZ') {
    stepCounter++;
    input = navMap[input][directions[directionInd]];
    directionInd = directionInd === directions.length - 1 ? 0 : directionInd + 1;
}

console.log(stepCounter);

File: 2021/aoc-day8-p1.py
# Put comma separated input here:
input1 = ""

input1 = input1.split(",")
count = 0
def z(stringy):
    if len(stringy) == 4 or len(stringy) == 2 or len(stringy) == 3 or len(stringy) == 7:
        return True
    else:
        return False
for x in input1:
    y = x.split(" ")
    if z(y[-1]):
        count+=1
    if z(y[-2]):
        count+=1
    if z(y[-3]):
        count+=1
    if z(y[-4]):
        count+=1
print(count)
